### PR TITLE
Add Automated Testing section on read the docs documentation

### PR DESCRIPTION
**Pull request by MinsukJi-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

Add descriptions of CI/CD testing and AutoRT testing in the read the docs documentation.

### Issue(s) addressed

#793 

## Testing

No testing required as this PR only changes documentation files located in the doc/UsersGuide/source/ directory

## Dependencies

No dependencies

### CODE REVIEW

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA how about merging this w/o testing immediately after merging the UPP PR?

**Review by junwang-noaa — APPROVED** [review]

**Comment by junwang-noaa** [comment]
Sure, that's fine to me.

File: doc/UsersGuide/source/AutomatedTesting.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#. Once all build jobs are successful, the created executable files are stored as
+   artifacts in GHA
+
+#. A number of AWS EC2 instances are started
+
+#. Test jobs are started on Amazon cloud by downloading the hpc-stack Docker container,
+   the executable file from the build job, and the input-data Docker container
+
+#. When all tests are finished, EC2 instances are stopped. Test results are reported
+   on GitHub
```
Comment (BrianCurtis-NOAA) [inline doc/UsersGuide/source/AutomatedTesting.rst]: Should add periods at the end of the `*` and `#.` sections

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by MinsukJi-NOAA** [comment]
This PR is ready to go.

**Comment by climbfuji** [comment]
I think we can merge this now.

**Pull request merged**